syntax="proto3";

service Predictor {
  // Predicts the decision for the provided instances. 
  rpc Predict(PredictRequest) returns (PredictResponse) {}
}

message History {
  repeated Item items = 1;
}

message Item {
  string dt = 1;
  int32 open = 2;
  int32 high = 3;
  int32 low = 4;
  int32 close = 5;
  int32 volume = 6;
}

message PredictRequest {

  History x_history=1;
  History y_history=2;
}

message PredictResponse {
  // X: 0.7, Y: 0.2, NOP: 0.1
  map<string, float> actions = 1;
}

